code/game/machinery/gym.dm

```diff
@@ -0,0 +1,152 @@
+/obj/machinery/gym
+	name = "Advanced Arcade Machine"
+	desc = "Links with your brain to reduce lag to minimum. Now, success really does depend only on your skill!"
+	description_info = "Exercise machines can be used to increase your stats, either permanently, by using a rest point, or temporary."
+	icon = 'icons/obj/machines/gym.dmi'
+	icon_state = "vigilance"
+
+	var/stat_used = STAT_VIG //STAT_TGH, STAT_ROB, STAT_VIG, STAT_COG, STAT_MEC, STAT_BIO
+	var/mob/living/carbon/human/occupant
+	var/unlocked = FALSE
+
+	density = TRUE
+	anchored = TRUE
+	circuit = null //cannot be deconstructed
+
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 60
+	active_power_usage = 400
+
+/obj/machinery/gym/robustness
+	name = "Interim Resistive Exercise Device"
+	desc = "This device uses a system of vacuum tubes and flywheel cables to simulate the process of free weight exercises that increase your strength. Are those barbells decorative...?"
+	icon_state = "robustness"
+
+	stat_used = STAT_ROB
+
+/obj/machinery/gym/toughness
+	name = "Total Resistance Punch Machine"
+	desc = "Whatever the reason behind creating this thing is, experiencing the life of a punching bag really helps you become tougher."
+	icon_state = "toughness"
+
+	stat_used = STAT_TGH
+
+/obj/machinery/gym/power_change()
+	..()
+	if(stat & BROKEN)
+		update_icon()
+
+/obj/machinery/gym/emag_act(remaining_charges, mob/user, emag_source)
+	emagged = TRUE
+
+/obj/machinery/gym/Destroy()
+	if(occupant)
+		occupant.ghostize(0)
+		occupant.gib()
```

Comment:
Hmm, we might need to tweak that later so that people don't stuff others in gym machines or something, just to destroy it immediately to gib the victim

Reply:
with a hatton or an admin verb? what's stopping them from doing so with just a hatton?

Reply:
Hatton does alright damage, but doesn't gib.